schemas/string_interpolation/transforms/generate_date_range.json

```diff
@@ -0,0 +1,74 @@
+{
+  "$id": "https://eq.ons.gov.uk/string_interpolation/transforms/generate_date_range.json",
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "generate_date_range": {
+    "description": "A transform for generating the start and end date of a date range from a reference date, offset number of weeks, number of days in range and an optional first day of the week.",
+    "type": "object",
+    "properties": {
+      "transform": {
+        "type": "string",
+        "enum": ["generate_date_range", "format_date_range_pair"]
+      },
+      "arguments": {
+        "type": "object",
+        "properties": {
+          "reference_date": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
+              {
+                "$ref": "https://eq.ons.gov.uk/common_definitions.json#/date_value"
+              }
+            ]
+          },
+          "offset_full_weeks": {
+            "oneOf": [
+              {
+                "$ref": "https://eq.ons.gov.uk/string_interpolation/definitions.json#/value_sources"
+              },
```

Comment:
Adding value sources into these inputs extends the scope as it is now within the current requirements. Happy to keep as it gives us the flexibility, just need to ensure tests these in runner with value sources as well as literal ints etc.

However, we would need to add additional python validation, i.e when an `answer` source is used, we need to validate that only a `Number` type is used and that it has a min value of `1`. etc. Same applies for `days_in_range`.

Reply:
What would be the alternative - isn't the `started_at` date destined to be a valuesource?

Reply:
Yes `started_at` will be a value source, but I don't follow how your comment relates to my initial comment?. My suggestion was if we support value sources for offset full weeks and days in range, they will need to be checked to ensure type is as expected, as you wouldn't want an text field answer value source for days in range. Given, we only have the requirement to make `reference_date` dynamic, these could be simplified and set to `integer` only which we can then expand if there is required to drive the offsets etc from a dynamic source.